### DISCUSSION TITLE
MathML Associated Files Examples.

### DISCUSSION THREAD

**Discussion by davidcarlisle** [latex3/tagging-project#56] [Ideas]
**Updated versions of the examples discussed here are now available in a larger [collection of PDF/UA-2 examples](https://github.com/latex3/tagging-project/discussions/72)**

----

Some examples showing initial attempts at tagging LaTeX math. Each math expression is tagged with two associated files, one with the original LaTeX source fragment, and one with MathML.  No explicit tagging markup needs to be added to the document body.

The PDF are produced using lualatex-dev.  Currently the TeX source are not shown but uncompressed PDF are provided here to allow comment on the PDF tagging structure used and to allow testing.

The files have been tested with [ngpdf](https://ngpdf.com/) and nvda+mathcat screen reader via a develpment build of [the foxit pdf reader](https://github.com/foxitsoftware/PDF_UA-2)

**Comment by davidcarlisle** [comment]
### Example 1

A small example, including inline math and a simple displayed equation.  The MathML conversion in this case was done by hand, including use of `intent="absolute-value($x)"` to ensure that `|x|` is read as "absolute value of x"

[mathml-AF-ex1.pdf](https://github.com/latex3/tagging-project/files/14390334/mathml-AF-ex1.pdf)



Tagged PDF and tagged PDF with TeX source fragments in /Alt



[mathml-AF-ex1-alt.pdf](https://github.com/latex3/tagging-project/files/14232608/mathml-AF-ex1-alt.pdf)

**Comment by davidcarlisle** [comment]
### Example 2

A larger example including math alignments from `amsmath` 


[mathml-AF-ex2.pdf](https://github.com/latex3/tagging-project/files/14390336/mathml-AF-ex2.pdf)

**Comment by davidcarlisle** [comment]
### Example 3

A re-creation in LaTeX of a test document originally provided by Neil Soiffer and produced in Word to test the nvda/mathcat readings.

The MathML conversions here were mostly done using tex4ht with some hand edits.


[Sample-AF-Math-LaTeX.pdf](https://github.com/latex3/tagging-project/files/14390339/Sample-AF-Math-LaTeX.pdf)

**Comment by davidcarlisle** [comment]
### amsmath User Guide (amsldoc)

A tagged version of the amsmath user guide. this is a much larger document than the above example (48 pages) and contains many non-typical math fragments as it is demonstrating features rather than showing normal math expressions.  The MathML conversion is a mixture of tex4ht conversion and hand written MathML.



[amsldoc-tagged.pdf](https://github.com/latex3/tagging-project/files/14390311/amsldoc-tagged.pdf)

**Comment by davidcarlisle** [comment]
### ArXiv Example 1

A larger example (276 math expressions) from a paper recently submited to arXiv under a [CC](https://creativecommons.org/licenses/by/4.0/) Licence permitting re-use such as this experiment, This document is available under the same licence.

All recent arXiv submissions  of LaTeX papers now include a (LaTeXML generated) HTML version including MathML conversions of all LaTeX math. This allows an automatic pipeline to produce tagged PDF including MathML associated files.

* Download the TeX source and html versions from arXiv
* add `\DocumentMetadata` to the TeX source to enable tagging and save as UTF-8.
* This document also required correction to the abstract markup and an adjustment to not load amsthm which is currently not supported
* Use an XSLT script to extract all MathML, simplifying the extracted MathML by removing content mathml annotations and merge html tables into single mtable expressions.
* run lualatex-dev to incorporate the mathml as an associated file on each Formula.

The table merging is required as LaTeXML uses html tables with a `<math>` element in each cell to represent amsmath alignments (which simplifies equation number positioning) but here we want a single `<math><mtable>` to enable MathCat to read the alignment.

Original sources 

https://arxiv.org/abs/2401.05361

Attached tagged PDF



[2401.05361v1-tagged.pdf](https://github.com/latex3/tagging-project/files/14390343/2401.05361v1-tagged.pdf)

**Comment by davidcarlisle** [comment]
### ArXiv Example 2

A second arXiv example from a recent (January 2024) submission under same CC licence as the previous example and built using essentially the same pipeline.

Original Sources

https://arxiv.org/abs/2401.09436

Attatched tagged PDF


[2401.09436v1-tagged.pdf](https://github.com/latex3/tagging-project/files/14390346/2401.09436v1-tagged.pdf)

**Comment by davidcarlisle** [comment]
### ArXiv Example 3

Another example, this time submitted to the Mathematical Physics arXiv rather than Mathematics. Again under under a [CC](https://creativecommons.org/licenses/by/4.0/) Licence .

This was again produced by extracting the MathML from the supplied html version.

Original Sources

https://arxiv.org/abs/2401.09965

Attatched tagged PDF


[2401.09965v1-tagged.pdf](https://github.com/latex3/tagging-project/files/14390351/2401.09965v1-tagged.pdf)

**Comment by bdoubrov** [comment]
The file amldoc.pdf from https://github.com/latex3/tagging-project/discussions/56#discussioncomment-8108785 has malformed structure tree: at least one Link structure element (page 13, object 1579) references itself as a parent (P entry in the structure element dictionary). This causes Acrobat to crash on this page (at least on my machine).

![image](https://github.com/latex3/tagging-project/assets/10971979/a0a13ed7-0230-405e-a2b9-191affde0096)

**Reply by davidcarlisle** [reply]
@bdoubrov  this looks like an issue with `\mathchoice` handling that had been fixed in development code. We have renewed all 6 example PDF generated from the latest tagging code. We think this error should be fixed. Note I also changed the name of the amsldoc example to amsldoc-tagged so as not to clash with the public original document

**Comment by davidcarlisle** [comment]
@bdoubrov  Curently in ngpdf all formulae are mapped to html figure, with 40pt margin.
the display of inline math is improved if you add
```
figure {margin:0}
```
in the css option in the editor


default:

![image](https://github.com/latex3/tagging-project/assets/1268738/b3baea1f-af91-46d5-9287-8f459bb9a690)


with CSS applied:

![image](https://github.com/latex3/tagging-project/assets/1268738/596197b8-84dc-4aef-a70f-ac82afd19cfa)
